### ISSUE TITLE
Feature request: add GetParentPool() to vmaGetAllocationInfo result?

### ISSUE THREAD

**Issue by danginsburg** [GPUOpen-LibrariesAndSDKs/VulkanMemoryAllocator#361]
Is there any reason not to include the parent pool in the result of `vmaGetAllocationInfo`?

The use case I have in mind is that I create custom pool(s) for some allocations and when all of the allocations in a pool are free'd, I want to free the custom pool.  When I perform the free, unless I keep a mapping of VmaAllocation -> VmaPool, I don't know whether my allocation is from a custom pool (nor which one).  If I had this info, I could quickly check if the pool was empty and then free it at this point.

Maybe I'm missing something but I don't see that info currently available.  I've locally hacked up VMA to provide this info, I can provide a PR for this if you like.

**Comment by danginsburg** [comment]
I understand not wanting feature creep, but I'm not sure that's a fair characterization.  The pool is already kept in the metadata in VMA, it's just adding it to an existing query.  It lets me write code like this without having to keep track of which allocations came from a custom pool and which didn't:

```
	VmaAllocationInfo allocationInfo = {};
	vmaGetAllocationInfo( m_pVmaAllocator, pVmaAllocation, &allocationInfo );
	vmaDestroyBuffer( m_pVmaAllocator, pBuffer, pVmaAllocation );
	if ( allocationInfo.pool != VK_NULL_HANDLE )
	{
		VmaFreeCustomPoolIfEmpty( allocationInfo.pool );
	}
````

It's about two lines of code, plus a comment.

**Comment by adam-sawicki-a** [comment]
It would be also about extending the structure `VmaAllocationInfo` and the function `vmaGetAllocationInfo`, which I want to be simple and work fast.

**Comment by danginsburg** [comment]
This is the diff I'm proposing, copying one VkPool pointer is going to slow it down?

```
1395a1396,1398
>
>       // https://github.com/GPUOpen-LibrariesAndSDKs/VulkanMemoryAllocator/issues/361
>       VmaPool VMA_NULLABLE pool;
15300a15304
>       pAllocationInfo->pool = hAllocation->GetParentPool();

**Comment by adam-sawicki-a** [comment]
Adding a new member to the structure that has 7 members means extending it by a large percent, which may impact the performance if the function is called thousands of times per frame or thousands of these structures are kept in an array.

Please feel free to maintain your own fork with this change if needed.

Can you please elaborate about your use case - why you need multiple custom pools? Custom pools are often over-used where default pools would be enough.

**Comment by danginsburg** [comment]
> Can you please elaborate about your use case - why you need multiple custom pools? Custom pools are often over-used where default pools would be enough.

The use case I have is that I have certain allocations that I am not able to move, namely the SBT and some other image allocations that are complex to relocate because of the way they are referenced.  What I had been doing is when defrag runs I would mark those `VMA_DEFRAGMENTATION_MOVE_OPERATION_IGNORE` but that was causing problems:

1. For performance reasons, I limit the defrag `maxAllocationsPerPass` and `maxBytesPerPass`, but sometimes all of the allocations in a pass would get marked ignore and this would cause the defrag to stop.  That is because here it does not mark the result as `VK_INCOMPLETE` (https://github.com/GPUOpen-LibrariesAndSDKs/VulkanMemoryAllocator/blob/1add1df154b2a05a6c7152407dcdcae30d983456/include/vk_mem_alloc.h#L13200).  I did have a local workaround for that as well, but then looking at memvis dumps I was also seeing that even if I worked around that there would be some large default pools never freed because they had some of these immovable allocations.  So that is why I'm using a custom pool.
2. Another thing I'm thinking about using custom pools for is that as far as I can tell the memory priority is only used for dedicated allocations, so it's not possible to specify a priority per default pool and search for allocations only by those priorites.
